### ISSUE TITLE
Missed "can only recur in a tail position" warning

### ISSUE THREAD

**Issue by NoahTheDuke** [clj-kondo/clj-kondo#2817]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

v2026.04.15

**macro usage**

N/A

**platform**

N/A

**editor**

N/A

**problem**

```clojure
(defn hello [a]
  [(recur (inc a))])
```

should warn for recur in a non-tail position. When loading the file or evaluating the form, Clojure throws

```
Syntax error (UnsupportedOperationException) compiling recur at (.../example.clj).
Can only recur from tail position
```

**repro**

```
$ echo '(defn hello [a] [(recur (inc a))])' | clj-kondo --lint -
linting took 17ms, errors: 0, warnings: 0
```

**config**

N/A

**expected behavior**

It should warn for this behavior.

**Comment by NoahTheDuke** [comment]
Thanks!
